### PR TITLE
 Creating ACM backplane permissions for LP-SRE.

### PR DESCRIPTION
**Pull request by Nanyte25**
### What type of PR is this?
_(bug/feature/cleanup/documentation)_

### What this PR does / why we need it?
 This PR is for backplane access for lpsre for the ACM service 
### Which Jira/Github issue(s) this PR fixes?

_Fixes #_

https://issues.redhat.com/browse/MTSRE-814
### Pre-checks (if applicable):
- [ *] Tested latest changes against a cluster
- [ ] Included documentation changes with PR
- [ ] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

### CODE REVIEW

**Comment by Nanyte25** [comment]
@jessesarn can you review

File: deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml [hunk 1]
```diff
@@ -0,0 +1,959 @@
...
+  - ''
+  resources:
+  - pods
+  - services
+  - services/finalizers
+  - endpoints
+  - persistentvolumeclaims
+  - events
+  - configmaps
+  - secrets
```
Comment (jewzaam) [inline deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml:441]: What secrets are available in the `redhat-open-cluster-management` namespace?

File: deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml [hunk 2]
```diff
@@ -0,0 +1,959 @@
...
+  - serviceaccounts
+  - namespaces
+  - nodes
```
Comment (jewzaam) [inline deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml:444]: This is bound to the `redhat-open-cluster-management` namespace so these resources do not make sense to include.

File: deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml [hunk 3]
```diff
@@ -0,0 +1,959 @@
...
+  verbs:
+  - delete
```
Comment (jewzaam) [inline deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml:446]: Why delete and not create?
And generally why delete?
Elevated permissions would be preferred for such operations, but I'd like to hear the use cases to understand if justified for the ACM addon.

File: deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml [hunk 4]
```diff
@@ -0,0 +1,959 @@
...
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - '*'
+  verbs:
+  - '*'
```
Comment (jewzaam) [inline deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml:517]:
```suggestion
  - get
  - list
  - watch
```
This is managed in MCC, the ability to change RBAC should not be granted directly.

File: deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml [hunk 5]
```diff
@@ -0,0 +1,959 @@
...
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - '*'
```
Comment (jewzaam) [inline deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml:545]:
```suggestion
```
Hard no on this at global scope.  This is bound to namespace right now but we do not want to accidentally allow changes to CRDs.  Elevated permissions must be used for such operations.

File: deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml [hunk 6]
```diff
@@ -0,0 +1,959 @@
...
+  - manifestworks
+  verbs:
+  - '*'
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - '*'
+  - infrastructures
+  verbs:
+  - '*'
```
Comment (jewzaam) [inline deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml:584]:
```suggestion
  - get
  - list
  - watch
```
These should not be direclty edited typically.  Elevated permissions of course can get access.

File: deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml [hunk 7]
```diff
@@ -0,0 +1,959 @@
...
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - certmanager.k8s.io
+  resources:
+  - '*'
+  verbs:
+  - '*'
```
Comment (jewzaam) [inline deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml:598]: Why is this required?

File: deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml [hunk 8]
```diff
@@ -0,0 +1,959 @@
...
+  - list
+  - watch
+- apiGroups:
+  - ''
+  resources:
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
```
Comment (jewzaam) [inline deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml:831]: Again, why is secret needed and what are in the secrets in the namespace?

**Review by jewzaam — CHANGES_REQUESTED** [review]
/hold

File: deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml (hunk 1 above)
Comment (Nanyte25) [reply deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml]: removed secret and namespace and nodes

File: deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml (hunk 4 above)
Comment (Nanyte25) [reply deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml]: @jewzaam I have removed all star permissions from cluster roles

File: deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml (hunk 8 above)
Comment (Nanyte25) [reply deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml:798]: removed `secrets` from permissions

Comment (Nanyte25) [reply deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml:798]: removed secret resources

Comment (Nanyte25) [reply deploy/backplane/lpsre/acm/02-acm-lpsre-admins-project.ClusterRole.yaml:798]: @jewzaam please review updated PR

**Comment by Nanyte25** [comment]
@jewzaam closing this out in favour of this PR [here](https://github.com/openshift/managed-cluster-config/pull/1455)
